crates/matrix-sdk-base/src/session.rs

```diff
@@ -39,18 +40,70 @@ use serde::{Deserialize, Serialize};
 pub struct Session {
     /// The access token used for this session.
     pub access_token: String,
+    /// The token used for [refreshing the access token], if any.
+    ///
+    /// [refreshing the access token]: https://spec.matrix.org/v1.3/client-server-api/#refreshing-access-tokens
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub refresh_token: Option<String>,
     /// The user the access token was issued for.
     pub user_id: OwnedUserId,
-    /// The ID of the client device
+    /// The ID of the client device.
     pub device_id: OwnedDeviceId,
 }
```

Comment:
I'm not sure I understand. Afaik, `Session` is not stored in our stores, not by the SDK anyways, it's only stored in memory and it's the SDK user's job to persist it somewhere. If they stored it somehow in the store there's no way for us to know how and where.

To mitigate this we could make this type not (de)serializable so users are forced to implement it themselves and to notice that the type changed when they update the SDK. Alternatively, we could introduce some kind of versioning for this type, ie make it an enum with variants `::V1(SessionV1)` and `::V2(SessionV2)` with methods to convert v1 to v2 and deprecate v1.

Reply:
Can't we just put a `default` field attribute on the refresh token instead?

Reply:
Afaik it's not needed for `Option`s. If a field is missing, it should be `None`. So the previous type should properly deserialize to this type if the user used it's `serde::Deserialize` implementation.

Reply:
`Option`s have an implicit `serde(default)`, you only need to specify it in some edge cases (I think when you also use `with` / `deserialize_with`).

Reply:
So we agree that there's no need for a migration?

Reply:
Right, using serde, we are ignorant about missing fields (mostly) and we are only using serde_json in our internal stores, so this can go on without migration. Only if we renamed fields this would be a problem.